tcl/register.tcl

```diff
@@ -2,46 +2,53 @@ namespace eval qc {
     namespace export register validate
 }
 
-proc qc::register {method path proc_args proc_body} {
+proc qc::register {args} {
     #| Register a URL handler.
-    set method [string toupper $method]
-    namespace eval ::${method} {}
-    set proc_name "::${method}::$path"
-    {*}[list proc $proc_name $proc_args $proc_body]
-
-    # Separate arg names and default values
-    set args {}
-    set defaults {}
-    foreach arg $proc_args {
-        if {[llength $arg] == 2} {
-            lappend args [lindex $arg 0]
-            dict set defaults [lindex $arg 0] [lindex $arg 1]
-        } else {
-            lappend args $arg
-        }
+    if { [llength $args] < 2 || [llength $args] > 4 } {
+        return -code error "Usage: qc::register method path ?args? ?body?"
     }
 
-    # Update the handlers nsv array.
-    qc::nsv_dict set handlers $method $path proc_name $proc_name
-    qc::nsv_dict set handlers $method $path args $args
-    qc::nsv_dict set handlers $method $path body $proc_body
-    qc::nsv_dict set handlers $method $path defaults $defaults    
-
-    if {$method eq "GET"} {
-        set http_methods [list HEAD GET]
+    set method [string toupper [lindex $args 0]]
+    set path [lindex $args 1]
+    # Add to registered nsv array
+    if { [qc::nsv_dict exists registered $method] } {
```

Comment:
No reason. I think it was initially written this way because it was written before `qc::registered` was made. Updating.